cloudbridge/cloud/providers/azure/services.py

```diff
@@ -1037,6 +1042,11 @@ def get_or_create_default(self, zone):
     def delete(self, subnet):
         subnet_id = subnet.id if isinstance(subnet, Subnet) else subnet
         self.provider.azure_client.delete_subnet(subnet_id)
+        network = subnet._network
+        az_network = network._network
+        az_network.tags.pop(subnet.tag_name)
+        self._provider.azure_client.update_network_tags(
+            az_network.id, az_network)
```

Comment:
@almahmoud Was this caught in a test? If not, can we write a test that would catch it?

Reply:
It was caught when we hit the limit after running multiple tests. I can write one specifically for it, but it would be an Azure specific test, since there is no way of really checking for a label in OS or AWS cause the objects don't exist anymore.

Reply:
Feels to me this could also use a comment as to why it's necessary to update the network tags.

Reply:
Yes, I guess there's no real way to test this in a cloud agnostic way, so perhaps we need not bother. Agree that a comment would be useful.

Reply:
Added the comment in https://github.com/CloudVE/cloudbridge/pull/152/commits/24d5b1f0e7449422c03e975851232217d67881e4